File: GolangAlgorithm/2Round/Chapter3/moveZeroes_283/moveZeroes_283.go
package moveZeroes_283

func moveZeroes(nums []int) {
	// 0,1,0,3,12->1,3,12,0,0
	loc := 0
	for k, v := range nums {
		if v != 0 {
			nums[loc] = v
			if k != loc {
				nums[k] = 0
			}
			loc++
		}
	}
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Array_List/DoublePointer/Array/moveZeroes_283.java
package algo.tzashinorpu.SpecialSubject.Array_List.DoublePointer.Array;

public class moveZeroes_283 {
    //    Input: nums = [0,1,0,3,12]
    //    Output: [1,3,12,0,0]
    public void moveZeroes(int[] nums) {
        for (int i = 0, p = 0; i < nums.length; i++) {
            if (nums[p] == 0 && nums[i] != 0) {
                nums[p++] = nums[i];
                nums[i] = 0;
            }
            if (nums[p] != 0) p++;
        }
    }
}
